api.ts

```diff
@@ -144,25 +144,16 @@ export interface Block {
   view_change_proofs: string[];
 }
 
-export type Transaction =
-  | Tagged<"Committed", CommittedTransaction>
-  | Tagged<"Rejected", RejectedTransaction>;
-
-export interface CommittedTransaction {
-  hash: string
-  /**
-   * WIP zeroed
-   */
+export interface TransactionDTO {
+  hash: string;
   block_hash: string;
   payload: TransactionPayload;
   signatures: Signature[];
-}
-
-export interface RejectedTransaction extends CommittedTransaction {
-  /**
-   * List of serialized {@link @iroha2/data-model#TransactionRejectionReason}
+ /**
+   * If the transactionQueryResults in error, this field will contain the rejection reason as a string.
+   * If not, this field will not be present (undefined or null).
    */
-  rejection_reason: string;
+ rejection_reason?: string | null;
```

Comment:
Currently `rejection_reason` is `string | null | undefined`. I don't see a reason why two nullable states are possible here. Serialisation in Rust works definitely. Suggest ensuring which is the correct one, consider one of the following changes instead:

```suggestion
 rejection_reason: string | null;
```

```suggestion
 rejection_reason?: string;
```